### DISCUSSION TITLE
Native Image Committer and Community Meeting 2022-10-20

### DISCUSSION THREAD

**Discussion by christianwimmer** [oracle/graal#5256] [Show and tell]
List of all past and upcoming meetings: #3933

## New and Noteworthy

* Disallow Native Image API classes and image builder internals in the application: Sometimes projects falsely depend on svm.jar (or graal-sdk.jar) or transitively include svm classes in uber-jars. This also happens e.g. when a maven projects depends on svm.jar but does not specify it as a dependency of type provided.
  #5232

* AWT support: We are switching to dynamic linking of C libraries for all platforms (like it always was on Windows). Some PRs towards this goal:
  #5249
  
* Allow "parse once" also for images that include Truffle languages, i.e., never parse bytecode after the static analysis. This will, e.g., allow method inlining before static analysis also for images that include Truffle languages. Some PRs towards this goal:
  #5061
  #5212
  
* Fix unnecessary imprecision of the static analysis regarding `instanceof` and other type checks: This exposes tricky corner cases, reducing the reachability can expose bugs in the configuration of native images.
  #5224

* Improve the traceability of the static analysis: More PRs to come in the near future.
  #5246
* Work for universal shadow heap and allowing to initialize all classes at image build time:
  #5174
* #5176
  
* Monitoring / tools:
  #5172
  #5254
  #5117
  #4932

  
## Deep Dive

* Questions / discussions regarding recent OpenJDK announcements.

* We are working to improve the unsability of the native image tool (like error messages). What are the biggest pain points right now?

## Possible deep dive topics for next meeting

* Upcoming improvements to reflection configuration: distinguish between "method not found" and "method not registered for reflection".

Please send suggestions in the GraalVM Slack channel

**Comment by galderz** [comment]
> We are working to improve the unsability of the native image tool (like error messages). What are the biggest pain points right now?

I've just created https://github.com/oracle/graal/issues/5267 as an example of misleading errors you can get when running native image with track object instantiation flag.
